src/buffer_writer.c

```diff
@@ -57,19 +59,65 @@ bool buffer_writer_write_value(buffer_writer_t *buffer_writer, const void *value
     return true;
 }
 
-bool buffer_writer_write_uint8(buffer_writer_t *buffer_writer, uint8_t value)
+bool MUSTCHECK buffer_writer_write_buffer(buffer_writer_t *buffer_writer, const buffer_writer_t *srcbuf)
+{
+    if(NULL == srcbuf || NULL == srcbuf->buffer)
+        return false;
+    return buffer_writer_write_value(buffer_writer, srcbuf->buffer->data, srcbuf->write_offset);
```

Comment:
The idea was to append a `buffer_writer` to another `buffer_writer`. `write_offset` is what determines how much of the buffer has been used. So the function works, but if it isn't used, I guess it should be removed. The `buffer_write_write_*` naming scheme is also a bit awkward, perhaps we could come up with something easier.